reports/report_csv.py

```diff
@@ -0,0 +1,174 @@
+    topic, agree_disagree = row
+    count = counts_list[row]
+    region = kwargs['region']
+    topic = [t for t in MAJOR_TOPICS if t[0] == topic][0][1]
+    agree_disagree = 'Agree' if agree_disagree == 1 else 'Disagree'
+    writer.writerow({'Region': region, 'Topic': topic, 'Answer': agree_disagree, 'Count': count})
+
+def ws_93_csv(writer, counts_list, row, **kwargs):
+    topic, yes_no = row
+    count = counts_list[row]
+    region = kwargs['region']
+    topic = [t for t in MAJOR_TOPICS if t[0] == topic][0][1]
+    yes_no = 'Yes' if yes_no == 'Y' else 'No'
+    writer.writerow({'Region': region, 'Topic': topic, 'Answer': yes_no, 'Count': count})
+
+def ws_97_csv(writer, counts_list, row, **kwargs):
+    regions = kwargs['regions']
+    for region_answer in counts_list[row]:
+        answer, region = region_answer
+        agree_disagree = 'Agree' if answer == 1 else 'Disagree'
+        region = [x[1] for x in regions if x[0] == region][0]
+        count = counts_list[row][region_answer]
+        writer.writerow({'Region': region, 'Topic': row, 'Agree/Disagree': agree_disagree, 'Count': count})
+
+def ws_100_csv(writer, counts_list, row, **kwargs):
+    for answer_topic in counts_list[row]:
+        answer, topic = answer_topic
+        yes_no = 'Yes' if answer == 'Y' else 'No'
+        topic = [t for t in MAJOR_TOPICS if t[0] == topic][0][1]
+        count = counts_list[row][answer_topic]
+        writer.writerow({'Topic': topic, 'Medium': row, 'Yes/No': yes_no, 'Count': count})
+
+def ws_101_csv(writer, counts_list, row, **kwargs):
+    gender, topic = row
+    gender = get_gender(gender)
+    topic = [t for t in MAJOR_TOPICS if t[0] == topic][0][1]
+    count = counts_list[row]
+    writer.writerow({'Topic': topic, 'Gender': gender, 'Count': count})
+
+def ws_102_csv(writer, counts_list, row, **kwargs):
+    agree_disagree, topic = row
+    topic = [t for t in MAJOR_TOPICS if t[0] == topic][0][1]
+    agree_disagree = 'Agree' if agree_disagree == 1 else 'Disagree'
+    count = counts_list[row]
+    writer.writerow({'Topic': topic, 'Agree/Disagree': agree_disagree, 'Count': count})
+
+def ws_104_csv(writer, counts_list, row, **kwargs):
+    function, gender = row
+    function = [func[1] for func in FUNCTION if func[0] == function]
+    if function:
+        function = function[0][4:] # [4:] is used to remove the (number) part
+        gender = get_gender(gender)
+        count = counts_list[row]
+        writer.writerow({'Topic': kwargs['topic'], 'Gender': gender, 'Function': function, 'Count': count})
+
+def generate_csv(csv_name, fieldnames, counts_list, func, **kwargs):
+    filename = f'csv/{csv_name}.csv'
+    file_exists = os.path.isfile(filename)
+    with open(filename, 'a+') as csv_file:
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+        if not file_exists:
+            writer.writeheader()
+        for row in (counts_list):
+            func(writer, counts_list, row, **kwargs)
```

Comment:
Not sure what's happening here but can't we always recreate the file from scratch?

Reply:
The reason we don't want to always create it from scratch is how the data comes in. The `generate_dataset` method is called the same way the `tabulate` functions are being invoked so Incase we are going through all the regions, we don't want to override the previous region data but instead we want to append the data since it's in the same worksheet.

Reply:
Then `tabulate_dataset` , etc., is probably a better name @esirK . `generate_xxx` gives the impression you're generating it for that worksheet rather than for for one row of the sheet.